core/src/main/java/de/evoila/cf/broker/service/DeploymentServiceImpl.java

```diff
@@ -143,14 +146,20 @@ public ServiceInstanceResponse syncCreateInstance (ServiceInstance serviceInstan
 		else {
 			serviceInstanceRepository.deleteServiceInstance(serviceInstance.getId());
 
-			throw new ServiceBrokerException(
-					"Internal error. Service instance was not created. ID was: " + serviceInstance.getId());
+			throw new ServiceBrokerException("Internal error. Service instance was not created. ID was: " + serviceInstance.getId());
 		}
 
 		try {
 			createdServiceInstance = platformService.postProvisioning(createdServiceInstance, plan);
 		} catch (PlatformException e) {
+			try {
+				platformService.deleteServiceInstance(serviceInstance);
+				serviceInstanceRepository.deleteServiceInstance(serviceInstance.getId());
+			} catch (PlatformException e1) {
+				throw new ServiceBrokerException("Could not delete failed instance " +serviceInstance.getId() + " due to: ", e);
```

Comment:
Is this according the OSB spec? I am not sure if I like the auto removal of the service instance after a failed deployment. Perhaps it should be marked for deletion and cleaned up after a grace period? @yremmet what is your opinion on that?

Reply:
This is not specific to this PR but so implemented in all our SBs.
Who would benefit from a grace period?
- For the platform the service is not available.
- The service does not contain any data because the provisioning failed.

We would have to add complexity to deleted failed services and the failed instances would linger around and consume resoures.

The OSB spec doesn't specify what should happen with failed instances

Reply:
Ok, then I am fine with it.